### PR TITLE
Support Yamaha routers and firewalls

### PR DESCRIPTION
**Pull request by Tachashi**
Added support for Yamaha routers (RTX) and firewalls (FWX).
https://www.yamaha.com/products/en/network/

Test Results

```
(venv_an2.8.4) [centos@localhost tests]$ ./test_yamaha.sh
Starting tests...good luck:
======================================================== test session starts ========================================================
platform linux -- Python 3.6.8, pytest-5.2.0, py-1.8.0, pluggy-0.13.0 -- /home/centos/venv_an2.8.4/bin/python3.6
cachedir: .pytest_cache
rootdir: /home/centos/venv_an2.8.4/netmiko, inifile: setup.cfg
plugins: f5-sdk-3.0.21
collected 14 items                                                                                                                  

test_netmiko_show.py::test_disable_paging PASSED                                                                              [  7%]
test_netmiko_show.py::test_ssh_connect PASSED                                                                                 [ 14%]
test_netmiko_show.py::test_ssh_connect_cm PASSED                                                                              [ 21%]
test_netmiko_show.py::test_send_command_timing PASSED                                                                         [ 28%]
test_netmiko_show.py::test_send_command PASSED                                                                                [ 35%]
test_netmiko_show.py::test_send_command_textfsm SKIPPED                                                                       [ 42%]
test_netmiko_show.py::test_send_command_genie SKIPPED                                                                         [ 50%]
test_netmiko_show.py::test_base_prompt PASSED                                                                                 [ 57%]
test_netmiko_show.py::test_strip_prompt PASSED                                                                                [ 64%]
test_netmiko_show.py::test_strip_command PASSED                                                                               [ 71%]
test_netmiko_show.py::test_normalize_linefeeds PASSED                                                                         [ 78%]
test_netmiko_show.py::test_clear_buffer PASSED                                                                                [ 85%]
test_netmiko_show.py::test_enable_mode PASSED                                                                                 [ 92%]
test_netmiko_show.py::test_disconnect PASSED                                                                                  [100%]

========================================================= warnings summary ==========================================================
/home/centos/venv_an2.8.4/lib64/python3.6/site-packages/_pytest/cacheprovider.py:127
  /home/centos/venv_an2.8.4/lib64/python3.6/site-packages/_pytest/cacheprovider.py:127: PytestCacheWarning: cache could not write path /home/centos/venv_an2.8.4/netmiko/.pytest_cache/v/cache/stepwise
    self.warn("cache could not write path {path}", path=path)

/home/centos/venv_an2.8.4/lib64/python3.6/site-packages/_pytest/cacheprovider.py:127
  /home/centos/venv_an2.8.4/lib64/python3.6/site-packages/_pytest/cacheprovider.py:127: PytestCacheWarning: cache could not write path /home/centos/venv_an2.8.4/netmiko/.pytest_cache/v/cache/nodeids
    self.warn("cache could not write path {path}", path=path)

-- Docs: https://docs.pytest.org/en/latest/warnings.html
====================================================== short test summary info ======================================================
SKIPPED [1] /home/centos/venv_an2.8.4/netmiko/tests/test_netmiko_show.py:83: TextFSM/ntc-templates not supported on this platform
SKIPPED [1] /home/centos/venv_an2.8.4/netmiko/tests/test_netmiko_show.py:109: TextFSM/ntc-templates not supported on this platform
============================================ 12 passed, 2 skipped, 2 warnings in 31.96s =============================================
======================================================== test session starts ========================================================
platform linux -- Python 3.6.8, pytest-5.2.0, py-1.8.0, pluggy-0.13.0 -- /home/centos/venv_an2.8.4/bin/python3.6
cachedir: .pytest_cache
rootdir: /home/centos/venv_an2.8.4/netmiko, inifile: setup.cfg
plugins: f5-sdk-3.0.21
collected 7 items                                                                                                                   

test_netmiko_config.py::test_ssh_connect PASSED                                                                               [ 14%]
test_netmiko_config.py::test_enable_mode PASSED                                                                               [ 28%]
test_netmiko_config.py::test_config_mode PASSED                                                                               [ 42%]
test_netmiko_config.py::test_exit_config_mode PASSED                                                                          [ 57%]
test_netmiko_config.py::test_command_set PASSED                                                                               [ 71%]
test_netmiko_config.py::test_commands_from_file PASSED                                                                        [ 85%]
test_netmiko_config.py::test_disconnect PASSED                                                                                [100%]

========================================================= warnings summary ==========================================================
/home/centos/venv_an2.8.4/lib64/python3.6/site-packages/_pytest/cacheprovider.py:127
  /home/centos/venv_an2.8.4/lib64/python3.6/site-packages/_pytest/cacheprovider.py:127: PytestCacheWarning: cache could not write path /home/centos/venv_an2.8.4/netmiko/.pytest_cache/v/cache/stepwise
    self.warn("cache could not write path {path}", path=path)

/home/centos/venv_an2.8.4/lib64/python3.6/site-packages/_pytest/cacheprovider.py:127
  /home/centos/venv_an2.8.4/lib64/python3.6/site-packages/_pytest/cacheprovider.py:127: PytestCacheWarning: cache could not write path /home/centos/venv_an2.8.4/netmiko/.pytest_cache/v/cache/nodeids
    self.warn("cache could not write path {path}", path=path)

-- Docs: https://docs.pytest.org/en/latest/warnings.html
============================================= 7 passed, 2 warnings in 72.19s (0:01:12) ==============================================
```

Sample Code (Japanese)
https://qiita.com/tech_kitara/items/b2496f74c5a4e0aab174

### CODE REVIEW

**Comment by akira6592** [comment]
Yamaha routers are very popular in Japan. 
I hope it will be merged.

File: netmiko/ssh_autodetect.py [hunk 1]
```diff
@@ -167,6 +167,12 @@
         "priority": 99,
         "dispatch": "_autodetect_std",
     },
+    "yamaha": {
+        "cmd": "show environment",
+        "search_patterns": [r"RTX", r"NVR", r"FWX"],
```
Comment (ktbyers) [inline netmiko/ssh_autodetect.py]: This is probably too general...would probably need an additional word like "Yamaha" in each of these search patterns (obviously has to match the output the device sends back for "show environment").

File: netmiko/yamaha/yamaha.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+        output = ""
+        msg = (
+            "Failed to enter administrator mode. Please ensure you pass "
+            "the 'secret' argument to ConnectHandler."
+        )
+        if not self.check_config_mode():
+            self.write_channel(self.normalize_cmd(config_command))
+            try:
+                output += self.read_until_prompt_or_pattern(pattern=pattern)
+                self.write_channel(self.normalize_cmd(self.secret))
```
Comment (ktbyers) [inline netmiko/yamaha/yamaha.py]: I think this probably should just be moved into the `enable()` call. So you have some sort of an `administrator` mode that requires you pass a second password and you need those privileges to make configuration changes. I think that should be moved to the `enable()` call (i.e. that probably is more consistent to Netmiko behavior than having this embedded in the `config_mode` call).

I am also thinking if we do this, then we can eliminate a bunch of code and rely more on `super()` calls to the base class.

File: netmiko/yamaha/yamaha.py [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+            self.write_channel("\n")
+            time.sleep(1)
+            output = self.read_channel()
+            if "(Y/N)" in output:
+                self.write_channel("N")
+            self.write_channel("\n")
+            time.sleep(1)
+            self.write_channel("\n")
+            time.sleep(1)
+            output += self.read_until_pattern(pattern=pattern)
```
Comment (ktbyers) [inline netmiko/yamaha/yamaha.py]: Can you provide more details on what happens here in the CLI? i.e. show the SSH-CLI that correponds to what happens here?

**Comment by ktbyers** [comment]
Updated this PR here:

https://github.com/ktbyers/netmiko/pull/1658

**Pull request closed without merging**